### PR TITLE
Browser integration =)

### PR DESCRIPTION
**Pull request by dluciv**
Please consider this "browser integration" pull request. It allows to bookmark current page easily via userscript.

I suggest squashing it because of lots of unneeded history in my repository.

I also have a kind of bookamrklet, but it fails on local files (just as expected), which, I guess, are widely used with your tool (at least my own bookmarks.public aren't public at all =)), so I don't suggest it to be added. But let it be here in the comment =):

    var ba = document.createElement('a');
    ba.innerHTML = "&#x1f516;";
    ba.target = "_blank";
    ba.href = "__URL_OF_YOUR_REPO__/index.html" +
        "?op=bookmark" +
        "&title=" + encodeURIComponent(document.title) +
        "&url="   + encodeURIComponent(window.location.href)
    ;
    Object.assign(ba.style, {
      'color': 'black',
      'font-size': '1.2cm'
    });
    var bd = document.createElement('div');
    Object.assign(bd.style, {
      'position': 'absolute',
      'z-index': 16777271,
      'top': "" + (window.pageYOffset + 10) + "px",
      'left': '10px',
      'width': '2cm',
      'height': '2cm',
      'background-color': 'white',
      'border': 'solid 1px black',
      'vertical-align': 'middle',
      'text-align': 'center'
    });
    bd.appendChild(ba);
    document.body.appendChild(bd);

### CODE REVIEW

**Comment by skx** [comment]
I appreciate your interest and contribution, but I admit I'm not 100% sure what to do with it.

As you say this fails on local file bookmarks, which is what I use and which I suspect is the most common way to use it.  (So, for example, my repository is located at `~/Repos/personal/bookmarks`, and my browser homepage is set to `file:///home/skx/Repos/personal/bookmarks/index.html`).

You suggest not merging, which makes sense, but also I can see that it would be interesting and useful to others.

Perhaps the best way forward is for you to keep your merge request to yourself, and me to link to your repository in the `README.md` file.  For those that are adding online it would then be a little more discoverable, and I wouldn't have to worry about it not working in the common-case.  Does that seem sane?  Or do you have another idea?

**Comment by dluciv** [comment]
Hello, Steve,
sorry, I guess I wrote a confusing text there =).

This pull request actually works well and can be either merged directly or squashed (probably better way just to keep it in one commit) into your repository.

It contains:

1. Functionality to open index.html with URL parameters instructing it to fill new bookmark's fields.
2. Small userscript that can be used to automate it in the browser. It works well on local files (checked with Firefox and Chrome), because userscript can open `file:///` urls, while regular JavaScript is not allowed to do that.

Then what about bookmarklet. I started to write it before userscript, but then realized that it will not be able to work as it will have no right to open `file:///` urls.

So basically I suggest two ways (up to your decision surely):

- Squash-merge (may be with some further correction) this pull request and forget about this bookmarklet
- Same as above, but with bookmarklet in `README.md` for those who use `http(s)://` to access their bookmarks (although I am not sure there are a lots of them) and prefer bookmarklet to userscript.

Or surely any other ideas from you =)

**Comment by skx** [comment]
One thing that does occur to me is that you could change this:

      ba.href = "__URL_OF_YOUR_REPO__/index.html" +
        "?op=bookmark" +
         "&title=" + encodeURIComponent(document.title) +
         "&url="   + encodeURIComponent(window.location.href)
       ;

To this:


      ba.href = "__URL_OF_YOUR_REPO__/index.html" +
        "#?op=bookmark" +
         "&title=" + encodeURIComponent(document.title) +
         "&url="   + encodeURIComponent(window.location.href)
       ;

That would allow a local `file://` to receive parameters - providing the tab wasn't already open - and populate the form.

That might mean we can make it work in either case..?

**Comment by dluciv** [comment]
No, actually we have different problem here.

`file://` URL receives both `?` and `#` parameters very well once opened. The problem is in opening `file://` URL itself.

Take a look on https://jsfiddle.net/dluciv/tbdvdL73/ — `file://` link is neither opened by clicking nor programmatically when referenced from `http(s)://`. `window.open`, `location = `, `browser.navigate`, etc. will also fail.

Userscripts and extensions can do it, while JS on the page and bookmarklets can not.

**Comment by skx** [comment]
You're right, thanks for the update.

If you give me a day or two to read over all the changes, some are obviously fine, some are unrelated (e.g. the scroll into view - seems reasonable, but unrelated really) then I'll merge.

I'll need the README to be updated with details of how to set this up and limitations too.

File: README.md [hunk 1]
```diff
@@ -81,6 +81,13 @@ The **Add Bookmark** form allows you to add new bookmark with name, link and com
 
 When moving your mouse over an existing bookmark, you can either click *recycle icon* to delete it or *pencil icon* to edit it.  The editing action works just in the same way as adding a new bookmark.
 
+### Browser Integration
+
+For the most of popular web browsers you can install a [userscript](https://en.wikipedia.org/wiki/Greasemonkey) from [here](add-bookmark.user.js).
+
+It will add userscript menu commands to setup URL of your `index.html` bookmark manager file and to bookmark current page.
+Last one will open bookmark manager with new bookmark title and URL filled in � very helpful.
+
```
Comment (skx) [inline README.md]: i.e.  This just needs a little extra detail.  Something like:

    The intention is that when you click on your bookmark the `add bookmark` area will be populated with the appropriate link and title, such that it can be added quickly and easily.
    This will work if your bookmarks are hosted remotely, and accessed via a `http` or `https` URL, but unfortunately
     if you're accessing locally, via a `file://`-based prefix then it will not work.

I don't know, it'll need a bit of explanation anyway.

Comment (dluciv) [inline README.md]: Ok, I'll write more detailed description, thank you!

**Comment by dluciv** [comment]
Here it is.

I have updated readme, now suggesting both options (userscript and bookmarklet), hope description is now friendly enough =)

**Comment by skx** [comment]
I appreciate your patience and your updates.
